03-viz.Rmd

```diff
@@ -1,40 +1,528 @@
+your message to others. Once you have selected a broad type of visualization to use, you will have to refine it to suit your particular need.
+Some rules of thumb for doing this are listed below. They generally fall into two classes: you want to *make your
+visualization convey your message / answer the question more clearly*, and you want to *simplify the visualization / reduce visual noise*
+as much as possible.
+
+**Convey the message**
+
+- Make sure the visualization answers the question you have asked in the simplest and most plain way possible.
+- Use legends and labels so that your visualization is understandable without reading the surrounding text.
+- Ensure the text, symbols, lines, etc. on your visualization are big enough to be easily read.
+- Make sure the data are clearly visible; don’t hide the shape/distribution of the data behind other objects (e.g. a bar).
+- Make sure to use colourschemes that are understandable by those with colourblindness (a surprisingly large fraction of the 
+  overall population). For example, [colorbrewer.org](https://colorbrewer.org)
+  and the `RColorBrewer` R library provide the ability to pick such colourschemes, and you can check your visualizations after
+  you have created them by uploading to online tools such as the [colour blindness simulator](https://www.color-blindness.com/coblis-color-blindness-simulator/).
+- Redundancy can be helpful; sometimes conveying the same message in multiple ways reinforces it for the audience.
+
+**Minimize noise**
+
+- Use colours sparingly. Too many different colours can be distracting, create false patterns, and detract from the message. 
+- Be wary of overplotting. If your plot has too many dots or lines and it starts to look like a mess, then you need to do something different.
+- Only make the plot area (where the dots, lines, bars are) as big as needed. Simple plots can be made small.
+- Don't adjust the axes to zoom in on small differences. If the difference is small, show that it's small!
+
+
+## Creating visualizations with `ggplot2` 
+#### *Build the visualization iteratively* {-#my-section}
+
+This section will cover examples of how to choose and refine a visualization given a dataset and a question that you want to answer, 
+and then how to create the visualization in R using `ggplot2`.
+
+### The Mauna Loa CO2 dataset 
+ This dataset contains the atmospheric concentration of carbon dioxide (CO2, in parts per million) at the Mauna Loa research station in Hawaii 
+from the years 1959-1997. **Question:** Does the concentration of atmospheric CO2 change over time, and are there any interesting patterns to note?
+```{r 03-data-co2}
+# mauna loa carbon dioxide data 
+# (with some light wrangling to make it a dataframe with informative column names)
+co2_df <- data.frame(concentration = as.matrix(co2), date = time(co2))
```

Comment:
This is new syntax... Might be better to hide this in a script and have them read in data like we did in chapter 2. If you do not agree, at least use `tibble`, the tidyverse function, in place of `data.frame` (for the same reason as `read_csv` in place of `read.csv`.

Also, you need to load tidyverse in this chapter, otherwise knitting it alone breaks it. When you do this, make sure you suppress warnings and messages using `warning=FALSE, message=FALSE` in the chunk options.

Reply:
cleaned data in script, used `read_csv` to load in textbook